com.unity.multiplayer.mlapi/Tests/Runtime/Metrics/RPC/NetworkMetricsRpcTests.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using MLAPI.Metrics;
+using NUnit.Framework;
+using Unity.Multiplayer.NetworkProfiler;
+using Unity.Multiplayer.NetworkProfiler.Models;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace MLAPI.RuntimeTests.Metrics.RPC
+{
+    public class NetworkMetricsRpcTests
+    {
+        NetworkManager m_Server;
+        NetworkManager m_Client;
+        NetworkMetrics m_ClientMetrics;
+        NetworkMetrics m_ServerMetrics;
+
+        [UnitySetUp]
+        public IEnumerator SetUp()
+        {
+            if (!MultiInstanceHelpers.Create(1, out m_Server, out NetworkManager[] clients))
+            {
+                Debug.LogError("Failed to create instances");
+                Assert.Fail("Failed to create instances");
+            }
+
+            var playerPrefab = new GameObject("Player");
+            NetworkObject networkObject = playerPrefab.AddComponent<NetworkObject>();
+            playerPrefab.AddComponent<RpcTestComponent>();
+
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(networkObject);
+
+            m_Server.NetworkConfig.PlayerPrefab = playerPrefab;
+            m_Client = clients.First();
+
+            m_Client.NetworkConfig.PlayerPrefab = playerPrefab;
+
+            if (!MultiInstanceHelpers.Start(true, m_Server, clients))
+            {
+                Debug.LogError("Failed to start instances");
+                Assert.Fail("Failed to start instances");
+            }
+
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.WaitForClientsConnected(clients));
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.WaitForClientConnectedToServer(m_Server));
+
+            m_ClientMetrics = m_Client.NetworkMetrics as NetworkMetrics;
+            m_ServerMetrics = m_Server.NetworkMetrics as NetworkMetrics;
+        }
+
+        [UnityTearDown]
+        public IEnumerator TearDown()
+        {
+            MultiInstanceHelpers.Destroy();
+
+            yield return null;
```

Comment:
Does this actually need to yield or could it just be a regular [Teardown]?

Reply:
I used what was done for the NetworkVariable tests.